pydantic/_internal/_docs_extraction.py

```diff
@@ -0,0 +1,89 @@
+"""Utilities related to attribute docstring extraction."""
+from __future__ import annotations
+
+import ast
+import inspect
+import re
+import textwrap
+from typing import Any
+
+
+class DocstringVisitor(ast.NodeVisitor):
+    def __init__(self) -> None:
+        super().__init__()
+
+        self.target: str | None = None
+        self.attrs: dict[str, str] = {}
+        self.previous_node_type: type[ast.AST] | None = None
+
+    def visit(self, node: ast.AST) -> Any:
+        node_result = super().visit(node)
+        self.previous_node_type = type(node)
+        return node_result
+
+    def visit_AnnAssign(self, node: ast.AnnAssign) -> Any:
+        if isinstance(node.target, ast.Name):
+            self.target = node.target.id
+
+    def visit_Expr(self, node: ast.Expr) -> Any:
+        if isinstance(node.value, ast.Str) and self.previous_node_type is ast.AnnAssign:
+            docstring = inspect.cleandoc(node.value.s)
+            if self.target:
+                self.attrs[self.target] = docstring
+            self.target = None
+
+
+def _extract_source_from_frame(lines: list[str], cls_name: str) -> list[str] | None:
+    source: list[str] = []
+    frame = inspect.currentframe()
+
+    if frame is None:
+        return None
```

Comment:
What I mean is that the while loop below will be skipped and `source` will still be empty, and then `if not source: return None` will trigger, so it's the same either way.

On that note, you could also immediately `return source` instead of `break`.